core/src/proxy/block_engine_stage.rs

```diff
@@ -170,7 +242,155 @@ impl BlockEngineStage {
                 &banking_packet_sender,
                 &exit,
                 &block_builder_fee_info,
-                &CONNECTION_TIMEOUT,
+                &mut error_count,
+                &shredstream_receiver_address,
+            )
+            .await;
+        }
+
+        Self::connect_auth_and_stream(
+            endpoint,
+            &local_block_engine_config,
+            &block_engine_config,
+            &cluster_info,
+            &bundle_tx,
+            &packet_tx,
+            &banking_packet_sender,
+            &exit,
+            &block_builder_fee_info,
+            &Self::CONNECTION_TIMEOUT,
+        )
+        .await
+    }
+
+    async fn connect_auth_and_stream_autoconfig(
+        mut backend_endpoint: Endpoint,
+        local_block_engine_config: &BlockEngineConfig,
+        global_block_engine_config: &Arc<Mutex<BlockEngineConfig>>,
+        cluster_info: &Arc<ClusterInfo>,
+        bundle_tx: &Sender<Vec<PacketBundle>>,
+        packet_tx: &Sender<PacketBatch>,
+        banking_packet_sender: &BankingPacketSender,
+        exit: &Arc<AtomicBool>,
+        block_builder_fee_info: &Arc<Mutex<BlockBuilderFeeInfo>>,
+        error_count: &mut u64,
+        shredstream_receiver_address: &Arc<ArcSwap<Option<SocketAddr>>>,
+    ) -> crate::proxy::Result<()> {
+        let mut endpoint_discovery = BlockEngineValidatorClient::connect(backend_endpoint.clone())
+            .await
+            .map_err(|e| ProxyError::BlockEngineConnectionError(e.to_string()))?;
+        let endpoints = endpoint_discovery
+            .get_block_engine_endpoints(GetBlockEngineEndpointRequest {})
+            .await
+            .map_err(|e| ProxyError::BlockEngineConnectionError(e.to_string()))?
+            .into_inner();
+
+        const PING_COUNT: usize = 3;
```

Comment:
move to function

Reply:
the issue with moving into the ping fn is we want to run a single ping so we don't have to wait for each individual response for the set of 3. eg you are in FRA, pinging TYO, you dont want to wait 250ms for each ping (total of 750ms), rather we want them to be fired off concurrently to avoid holding up the connection startup loop

Reply:
yeah, mainly saying move the parallel pinging to a diff function. this one pretty beefy right now